kratos/geometries/coupling_geometry.h

```diff
@@ -0,0 +1,357 @@
+            << mpGeometries.size() << " geometries." << std::endl;
+
+        mpGeometries[Index] = pGeometry;
+    }
+
+    /**
+    * @brief ONLY for coupling_geometry. Not necessary in base class.
+    * @details Allows to enhance the coupling geometry, with another geometry.
+    * @param Index of the geometry part. 0->Master; 1->Slave
+     */
+    IndexType AddGeometryPart(GeometryPointer pGeometry)
+    {
+        KRATOS_DEBUG_ERROR_IF(mpGeometries[0]->Dimension() != pGeometry->Dimension())
+            << "Geometries of different dimensional size!" << std::endl;
+
+        IndexType new_index = mpGeometries.size();
+
+        mpGeometries.resize(new_index + 1);
+        mpGeometries[new_index] = pGeometry;
+
+        return new_index;
+    }
+
+    /**
+     * @brief The number of geometry part
+     * @return The number of geometry parts that this geometry contains.
+     */
+    SizeType NumberOfGeometryParts() const override
+    {
+        return mpGeometries.size();
+    }
+
+    ///@}
+
+    /**
+     * @brief Returns the domain size of the master geometry.
+     * @return The domaon size of the master geometry
+     */
+    double DomainSize() const override
+    {
+        return mpGeometries[0]->DomainSize();
+    }
+
+    /**
+     * @brief Returns the center of the master geometry.
+     * @return The center of the master geometry
+     */
+    Point Center() const override
+    {
+        return mpGeometries[0]->Center();
+    }
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override
+    {
+        return "Coupling geometry that holds a master and a set of slave geometries.";
+    }
+
+    /// Print information about this object.
+    void PrintInfo( std::ostream& rOStream ) const override
+    {
+        rOStream << "Coupling geometry that holds a master and a set of slave geometries.";
+    }
+
+    /// Print object's data.
+    void PrintData( std::ostream& rOStream ) const override
+    {
+        BaseType::PrintData( rOStream );
+        std::cout << std::endl;
+        rOStream << "    CouplingGeometry with " << mpGeometries.size() << " geometries.";
+    }
+
+    ///@}
+protected:
+
+private:
+    ///@name Private Member Variables
+    ///@{
+
+    std::vector<GeometryPointer> mpGeometries;
+
+    ///@}
+    ///@name Serialization
+    ///@{
+
+    friend class Serializer;
+
+    void save( Serializer& rSerializer ) const override
+    {
+        KRATOS_SERIALIZE_SAVE_BASE_CLASS( rSerializer, BaseType );
+        rSerializer.save("Geometries", mpGeometries);
+    }
+
+    void load( Serializer& rSerializer ) override
+    {
+        KRATOS_SERIALIZE_LOAD_BASE_CLASS( rSerializer, BaseType );
+        rSerializer.load("Geometries", mpGeometries);
+    }
+
+    CouplingGeometry() : BaseType() {}
+
+    ///@}
+    ///@name Private Friends
+    ///@{
+
+    template<class TOtherPointType> friend class CouplingGeometry;
```

Comment:
Are you sure this is necessary?

Reply:
I think this is necessary if the point type is changed and the geometry needs to be 'converted', but I am not sure. I just adapted it in the same way as the other geometries are using it.

Reply:
ok